LayoutKit/Layout.swift

```diff
@@ -82,6 +86,12 @@ public protocol Layout {
      TODO: figure out how to assert if inflexible layouts are compressed.
      */
     var flexibility: Flexibility { get }
+
+    /**
+     An identifier for the layout.
+     It is used to identify which views should be reused when animating from one layout to another.
```

Comment:
I'd add a bit more detail here, talking about how applying two layouts reuses views by this id. Maybe even rename it to viewReuseId?

Also somewhere we should describe what happens if there is a duplicate.